lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3093,6 +3093,86 @@ LogicalResult ConvertAtenOp<AtenBroadcastToOp>::matchAndRewrite(
       "unimplemented: broadcasts other than same rank or zero ranked tensor.");
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenGatherOp>::matchAndRewrite(
+    AtenGatherOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  // For easy understanding of this algorithm, I will comment the code with an
+  // exact example: torch.aten.gather (!torch.vtensor<[1,4,3],f32>,
+  // !torch.int-1, !torch.vtensor<[1,4,2],si64>)
+  // -> !torch.vtensor<[1,4,2],f32>
+  // https://gist.github.com/AmosLewis/2f18434397025211da4491735bcc6db6
+
+  // Not a tensor type.
+  auto input = adaptor.getSelf();
+  auto inputType = adaptor.getSelf().getType().dyn_cast<RankedTensorType>();
+  if (!inputType)
+    return rewriter.notifyMatchFailure(
+        op, "Only RankedTensorType input are currently supported");
+
+  auto index = adaptor.getIndex();
+  auto indexType = adaptor.getIndex().getType().dyn_cast<RankedTensorType>();
+  auto inputShape = makeShapeTorchCompatible(inputType.getShape());
+  auto indexShape = makeShapeTorchCompatible(indexType.getShape());
+  int paramsRank = inputShape.size();
+
+  // Check `index` and `input` param should have the same rank
+  if (indexType.getRank() != inputType.getRank()) {
```

Comment:
Don't you need to check if `indexType` is Null or not after the dyn_cast? Or if it's not going to be null ever then instead use `cast` and the `RankedTensorType indexType`.

Reply:
Need to add one more check before this
```
if (!indexType)
    return rewriter.notifyMatchFailure(
        op, "Only RankedTensorType input are currently supported");
```

Reply:
Done